pkg/apis/messaging/v1beta1/subscription_types.go

```diff
@@ -0,0 +1,147 @@
+// You can also perform an identity transformation on the incoming events by leaving
+// out the Subscriber and only specifying Result.
+//
+// The following are all valid specifications:
+// channel --[subscriber]--> reply
+// Sink, no outgoing events:
+// channel -- subscriber
+// no-op function (identity transformation):
+// channel --> reply
+type SubscriptionSpec struct {
+	// Reference to a channel that will be used to create the subscription
+	// You can specify only the following fields of the ObjectReference:
+	//   - Kind
+	//   - APIVersion
+	//   - Name
+	//  The resource pointed by this ObjectReference must meet the
+	// contract to the ChannelableSpec duck type. If the resource does not
+	// meet this contract it will be reflected in the Subscription's status.
+	//
+	// This field is immutable. We have no good answer on what happens to
+	// the events that are currently in the channel being consumed from
+	// and what the semantics there should be. For now, you can always
+	// delete the Subscription and recreate it to point to a different
+	// channel, giving the user more control over what semantics should
+	// be used (drain the channel first, possibly have events dropped,
+	// etc.)
+	Channel corev1.ObjectReference `json:"channel"`
+
+	// Subscriber is reference to (optional) function for processing events.
+	// Events from the Channel will be delivered here and replies are
+	// sent to a Destination as specified by the Reply.
+	// +optional
+	Subscriber *duckv1.Destination `json:"subscriber,omitempty"`
+
+	// Reply specifies (optionally) how to handle events returned from
+	// the Subscriber target.
+	// +optional
+	Reply *duckv1.Destination `json:"reply,omitempty"`
+
+	// Delivery configuration
+	// +optional
+	Delivery *eventingduckv1alpha1.DeliverySpec `json:"delivery,omitempty"`
+}
+
+// SubscriptionStatus (computed) for a subscription
+type SubscriptionStatus struct {
+	// inherits duck/v1 Status, which currently provides:
+	// * ObservedGeneration - the 'Generation' of the Service that was last processed by the controller.
+	// * Conditions - the latest available observations of a resource's current state.
+	duckv1.Status `json:",inline"`
+
+	// PhysicalSubscription is the fully resolved values that this Subscription represents.
+	PhysicalSubscription SubscriptionStatusPhysicalSubscription `json:"physicalSubscription,omitempty"`
+}
+
+// SubscriptionStatusPhysicalSubscription represents the fully resolved values for this
+// Subscription.
+type SubscriptionStatusPhysicalSubscription struct {
+	// SubscriberURI is the fully resolved URI for spec.subscriber.
+	SubscriberURI *apis.URL `json:"subscriberURI,omitempty"`
+
+	// ReplyURI is the fully resolved URI for the spec.reply.
+	ReplyURI *apis.URL `json:"replyURI,omitempty"`
+
+	// ReplyURI is the fully resolved URI for the spec.delivery.deadLetterSink.
+	DeadLetterSinkURI *apis.URL `json:"deadLetterSinkURI,omitempty"`
```

Comment:
the correct json should be deadLetterSinkUri`

Why is the channel `error` and this `deadLetter`?